src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.OpenFileTracker.cs

```diff
@@ -321,27 +319,20 @@ private void RefreshContextsForHierarchyPropertyChange(IVsHierarchy hierarchy)
                     {
                         if (subscribedHierarchy.Target.Key == hierarchy)
                         {
-                            RefreshContextForRunningDocumentTableCookie(cookie);
+                            var moniker = _runningDocumentTable.GetDocumentMoniker(cookie);
```

Comment:
if this is only place where _runningDocumentTable is used, I would add GetMoniker in runningDocumentTableEventSink so that raw runningdoc table is no longer needed. basically I would delegate all work to raw RDT to your new RDT helper.

Reply:
It is not the only place, it also deals with the hierarchical stuff in this file too.

@heejaechang I could just wrapper methods around all the methods used here, would that be worth doing?

In the open file tracker it uses-
```IsDocumentInitialized```, ```GetDocumentHierarchyItem```, ```GetDocumentMoniker```, ```IsMonikerValid```, ```GetDocumentCookie```

Reply:
I commented on the OpenFileTracker usages but I think some higher level operations might be useful: instead what much of this is trying to do is provide "get me a document by file name" or "get me all open documents".